packages/extension-commerce-bm-seo/src/setup-app.tsx

```diff
@@ -0,0 +1,181 @@
+const sliceInitializer: SliceInitializer<StoreSlice> = (set) => ({
+    count: 0,
+    increment: () => set((state) => ({count: state.count + 1})),
+    decrement: () => set((state) => ({count: state.count - 1}))
+})
+
+class CommerceBmSeo extends ApplicationExtension<Config> {
+    static readonly id = extensionMeta.id
+
+    /**
+     * Use this method to wrap or enhance your PWA-Kit application using [React higher-order components](https://legacy.reactjs.org/docs/higher-order-components.html).
+     * You can use this to add visual treatments to your application, change the props that are supplied to the application component
+     * or add things like providers and contexts to be used throughout your app.
+     */
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
+        const HOCs = [
+            // Optionally include state for this extension using `withApplicationExtensionStore`
+            (component: React.ComponentType<any>) =>
+                withApplicationExtensionStore(component, {
+                    id: extensionMeta.id,
+                    initializer: sliceInitializer
+                })
+        ]
+
+        return applyHOCs(App, HOCs)
+    }
+
+    /**
+     * This method is used to make changes to the PWA-Kit application routes. If your extension adds a new page to the application
+     * then you can add it to the router here. The routes passed to this method is an accrued list of routes that have been added
+     * from extensions applied before it. It is called during the `getRoutes` phase on both the server and the client.
+     *
+     * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
+     */
+    async getRoutesAsync({locals}: GetRoutesParams): Promise<RouteProps[]> {
+        let urlMapping
+
+        // TODO: do we need this check?
+        if (!locals.originalUrl) {
+            return Promise.resolve([])
+        }
+
+        // Make SEO GET Url Mapping API call
+        const urlSegment: string = locals.originalUrl.split('?')[0]
+        const config = this.getConfig()
+        const shopperSeo = await getShopperSeoClient(locals, config)
+        try {
+            urlMapping = await shopperSeo.getUrlMapping({
+                parameters: {urlSegment}
+            })
+        } catch (e) {
+            console.error(`Couldn't find mapping for given segement: ${urlSegment}`)
+        }
+
+        if (!urlMapping) {
+            return Promise.resolve([])
+        }
+
+        const {component, props} = getComponentForUrlMapping(
+            urlMapping,
+            config.resourceTypeToComponentMap
+        )
+        const requestURL = new URL(urlSegment, getAppOrigin(locals))
+        return Promise.resolve([
+            {
+                path: requestURL.pathname,
+                component,
+                componentProps: props,
```

Comment:
here we now pass the component's props as part of the route so it can be serialized

Reply:
@hajinsuha1 just an aside: sometimes I feel you have important comments made in Github PR that I think it's best if they're written as code comments instead. Such comments in Github would be easily lost/forgotten.